haze/src/skikoMain/kotlin/dev/chrisbanes/haze/HazeNode.kt

```diff
@@ -34,19 +36,38 @@ private const val SHADER_SKSL = """
   uniform shader noise;
 
   uniform vec4 rectangle;
+  uniform vec2 topLeftCornerRadius;
+  uniform vec2 topRightCornerRadius;
+  uniform vec2 bottomRightCornerRadius;
+  uniform vec2 bottomLeftCornerRadius;
   uniform vec4 color;
   uniform float colorShift;
 
   // https://www.iquilezles.org/www/articles/distfunctions2d/distfunctions2d.htm
-  float boxSDF(vec2 position, vec2 box) {
-      vec2 q = abs(position) - box;
-      return length(max(q,0.0)) + min(max(q.x,q.y),0.0);
+  float boxSDF(vec2 position, vec2 box, vec2 radius) {
+    float cornerRadius = max(radius.x, radius.y);
+    vec2 q = abs(position) - box + vec2(cornerRadius);
+    return length(max(q,0.0)) + min(max(q.x,q.y),0.0) - cornerRadius;
   }
 
   vec4 main(vec2 coord) {
     vec2 shiftRect = (rectangle.zw - rectangle.xy) / 2.0;
     vec2 shiftCoord = coord - rectangle.xy;
-    float distanceToClosestEdge = boxSDF(shiftCoord - shiftRect, shiftRect);
+    vec2 cornerRadius;
+    if (shiftCoord.x > shiftRect.x) {
```

Comment:
This can all be simplified with the `sdRoundedBox` function above.

Reply:
Done, It's much simpler now.